### PR TITLE
Deprecate context methods

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Active span management is going away soon in #1936 which involves
significant changes to the Context API.

These methods should not be used by most users (if you are using them
please let us know!) and are not backwards compatible with the changes
being made.

### CODE REVIEW

**Review by jd — DISMISSED** [review]

**Review by P403n1x87 — APPROVED** [review]

File: releasenotes/notes/context-df929fc8dfe148ba.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+prelude: >
+    Several deprecations have been made to `Context` as we prepare to move
```
Comment (P403n1x87) [inline releasenotes/notes/context-df929fc8dfe148ba.yaml:3]:
~~~
/home/circleci/project/docs/release_notes.rst:14: Spell check: deprecations: Several deprecations have been made to .
~~~
🙁

**Review by P403n1x87 — APPROVED** [review]

**Review by jd — APPROVED** [review]

File: releasenotes/notes/context-df929fc8dfe148ba.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  - |
+    `Context.add_span` is deprecated and will be removed in 0.50.
+  - |
+    `Context.close_span` is deprecated and will be removed in 0.50.
+  - |
+    `Context.get_current_span` is deprecated and will be removed in 0.50 please
+    use `Tracer.current_span` instead.
+  - |
+    `Context.get_current_root_span` is deprecated and will be removed in 0.50
+    please use `Tracer.current_root_span` instead.
```
Comment (brettlangdon) [inline releasenotes/notes/context-df929fc8dfe148ba.yaml:15]:
```suggestion
    Several deprecations have been made to ``Context`` as we prepare to move
    active span management out of this class.
deprecations:
  - |
    ``Context.add_span`` is deprecated and will be removed in 0.50.
  - |
    ``Context.close_span`` is deprecated and will be removed in 0.50.
  - |
    ``Context.get_current_span`` is deprecated and will be removed in 0.50 please
    use `Tracer.current_span` instead.
  - |
    ``Context.get_current_root_span`` is deprecated and will be removed in 0.50
    please use ``Tracer.current_root_span`` instead.
```

RST needs double ticks (I am pretty sure?)

Comment (Kyle-Verhoog) [reply releasenotes/notes/context-df929fc8dfe148ba.yaml:15]: Yeah I never know 😅

**Review by brettlangdon — DISMISSED** [review]

File: releasenotes/notes/context-df929fc8dfe148ba.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+prelude: >
+    Several deprecations have been made to ``Context`` as we prepare to move
+    active span management out of this class.
+deprecations:
+  - |
+    ``Context.add_span`` is deprecated and will be removed in 0.50.
```
Comment (majorgreys) [inline releasenotes/notes/context-df929fc8dfe148ba.yaml:7]: Missing deprecation for `clone`, otherwise looks good.

Comment (Kyle-Verhoog) [reply releasenotes/notes/context-df929fc8dfe148ba.yaml:7]: Ah good catch!

**Review by P403n1x87 — DISMISSED** [review]

File: releasenotes/notes/context-df929fc8dfe148ba.yaml [hunk 4]
```diff
@@ -3,8 +3,12 @@ prelude: >
     Several deprecations have been made to ``Context`` as we prepare to move
     active span management out of this class.
 deprecations:
+  - |
+    ``Context.clone`` is deprecated. It will not be required in 0.50.
   - |
     ``Context.add_span`` is deprecated and will be removed in 0.50.
+  - |
+    `Context.add_span` is deprecated and will be removed in 0.50.
```
Comment (brettlangdon) [inline releasenotes/notes/context-df929fc8dfe148ba.yaml:11]:
```suggestion
    ``Context.add_span`` is deprecated and will be removed in 0.50.
```

Comment (Kyle-Verhoog) [reply releasenotes/notes/context-df929fc8dfe148ba.yaml:11]: dang it

**Review by P403n1x87 — APPROVED** [review]
🟢 🟢 🟢 🟢 🟢 🟢 🟢

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**